File: Ruby/c_rank/judge_right_triangle.rb
m, n = gets.split(" ").map(&:to_i)

count = 0
m.times do |bottom|
  a = bottom
  # aが0の時はスキップ
  if a == 0
    next
  end
  n.times do |height|
    b = height

    # bが0の時はスキップ
    if b == 0
      next
    end

    # 三平方の定理を使用
    c = Math.sqrt(a**2 + b**2)

    # cが整数の時のみカウント
    if c - c.to_i == 0
      count += 1
    end
  end
end

puts count

File: java/c_rank/JudgeRightTriangle.java
package java.c_rank;

import java.util.Scanner;

public class JudgeRightTriangle {
  public static void main(String[] args) {
    Scanner sc = new Scanner(System.in);
    
    int bottom;
    int height;
    bottom = sc.nextInt();
    height = sc.nextInt();
    sc.close();

    int count = 0;

    for (int i = 1; i < bottom; i++) {
      for (int j = 1; j < height; j++) {
        double obliqueLine = Math.sqrt(Math.pow(i, 2) + Math.pow(j, 2));
        int intObliqueLine = (int)(obliqueLine);
        if (obliqueLine - intObliqueLine == 0) {
          count += 1;
        }
      }
    }
    
    System.out.println(count);
  }
}
